src/main/drivers/bus_i2c_stm32f30x.c

```diff
@@ -231,15 +231,17 @@ bool i2cWrite(uint8_t addr_, uint8_t reg, uint8_t data)
         }
     }
 
-    /* Send Register address */
-    I2C_SendData(I2Cx, (uint8_t) reg);
+    if (reg != -1) {
```

Comment:
The code was tested.
The first call to the `I2C_TransferHandling` generates slave address and a write bit. Since we set the `I2C_Reload_Mode`, we can write as many bytes thereafter. However, we eventually have to generate a stop condition, and the original code decided to program it between reg and data writes at line 253 with `I2C_AutoEnd_Mode`.

Here's the test code:
```
    // Test i2cWrite() and i2cRead()
    uint8_t tmpwbuf[1] = { 0x55 };
    uint8_t tmprbuf[2];

    // With sub-addressing
    i2cWrite(0x42, 0, tmpwbuf[0]);  // S 42(w) 00 55 P
    i2cRead(0x42, 0, 2, tmprbuf);   // S 42(w) 00 R 42(r) 00 01 P

    // Without sub-addressing
    i2cWrite(0x42, -1, tmpwbuf[0]); // S 42(w) 55 P     (Treated as reg)
    i2cRead(0x42, -1, 2, tmprbuf);  // S 42(r) 55 56 P  
```
And the bus show-off.
<img width="1175" alt="screen shot 2016-07-26 18 28 25" src="https://cloud.githubusercontent.com/assets/14850998/17133259/58148248-5360-11e6-8a5e-140ca052b0e5.png">